File: solutions/solution_01.py
def getInput():
    return open("./input/input_01.txt")

# Part 1
def part1():
    input = getInput()
    shifts = 0
    for shift in input:
        shifts = shifts + int(shift)
    return shifts

# Part 2
log = {"0" : True}
def part2(sum = 0):
    input = getInput()
    for shift in input:
        sum = sum + int(shift)
        if str(sum) in log:
            return sum
        else:
            log[str(sum)] = True
    return part2(sum)

File: 2017/solutions/solution_01.js
const fs = require("fs");
const input = fs.readFileSync(__dirname + "/../input/input_01.txt", "utf8")
  .trim();

// Part 1
function part1() {
  return input.split("").reduce((s, d, i, a) => s += (d == a[(i + 1) % a.length] ? parseInt(d, 10) : 0), 0);
}

// Part 2
function part2() {
  return input.split("").reduce((s, d, i, a) => s += (d == a[(i + a.length / 2) % a.length] ? parseInt(d, 10) : 0), 0);
}

module.exports = { part1, part2 }
